src/OmniSharp.MSBuild/ProjectFile/ProjectFileInfo.cs

```diff
@@ -142,13 +144,30 @@ public void SetProjectId(ProjectId projectId)
                 globalProperties.Add(PropertyNames.VisualStudioVersion, options.VisualStudioVersion);
             }
 
-            var collection = new ProjectCollection(globalProperties);
+            var collection = new ProjectCollection(globalProperties, new[] { new MSBuildLogForwarder(logger, diagnostics) }, ToolsetDefinitionLocations.Default);
 
+            // Evaluate the MSBuild project
             var project = string.IsNullOrEmpty(options.ToolsVersion)
                 ? collection.LoadProject(projectFilePath)
                 : collection.LoadProject(projectFilePath, options.ToolsVersion);
 
-            logger.LogInformation($"Using tools version: {project.ToolsVersion}");
+            var targetFramework = project.GetPropertyValue(PropertyNames.TargetFramework);
+            var targetFrameworks = PropertyConverter.ToList(project.GetPropertyValue(PropertyNames.TargetFrameworks), ';');
+
+            // If the project supports multiple target frameworks and specific framework isn't
+            // selected, we must pick one before execution. Otherwise, the ResolveReferences
+            // target might not be available to us.
+            if (string.IsNullOrWhiteSpace(targetFramework) && targetFrameworks.Count > 0)
+            {
+                // For now, we'll just pick the first target framework. Eventually, we'll need to
+                // do better and potentially allow OmniSharp hosts to select a target framework.
+                targetFramework = targetFrameworks[0];
```

Comment:
In the past we've exposed this on each request, but a dedicated endpoint would be ideal.

What about support in VScode?  Any UI that might be setup?